projects/plugins/boost/app/modules/cache/cache.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+class Boost_Cache {
+	public $key         = false;
+	public $request_uri = false;
+
+	public function __construct() {
+		$this->key();
+		$this->request_uri = isset( $_SERVER['REQUEST_URI'] )
+			? $_SERVER['REQUEST_URI'] // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized, WordPress.Security.ValidatedSanitizedInput.MissingUnslash
+			: false;
+
+		add_action( 'wp', array( $this, 'sanitize' ), 0 );
+	}
+
+	public function sanitize() {
+		if ( ! $this->request_uri ) {
+			return;
+		}
+
+		$this->request_uri = esc_url_raw( wp_unslash( $this->request_uri ) );
+	}
```

Comment:
🤔 Is this delayed until `wp_unslash` is available?

Reply:
Yeah, it runs on the "wp" action. It's possibly not needed, but in Super Cache I needed to keep a backup of it in case plugins messed around with it. I used it when storing meta information, and some checks.